src/test/java/com/android/volley/toolbox/DiskBasedAsyncCacheTest.java

```diff
@@ -0,0 +1,450 @@
+package com.android.volley.toolbox;
+
+import static org.hamcrest.Matchers.arrayWithSize;
+import static org.hamcrest.Matchers.emptyArray;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+
+import androidx.annotation.Nullable;
+import com.android.volley.AsyncCache;
+import com.android.volley.Cache;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.AsynchronousFileChannel;
+import java.nio.file.StandardOpenOption;
+import java.util.HashMap;
+import java.util.Random;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(sdk = 26)
+public class DiskBasedAsyncCacheTest {
+
+    private static final int MAX_SIZE = 1024 * 1024;
+
+    private DiskBasedAsyncCache cache;
+
+    private AsyncCache.OnCompleteCallback emptyCallback;
+
+    @Rule public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Rule public ExpectedException exception = ExpectedException.none();
+
+    @Before
+    public void setup() throws IOException {
+        emptyCallback =
+                new AsyncCache.OnCompleteCallback() {
+                    @Override
+                    public void onComplete() {
+                        // do nothing
+                    }
+                };
```

Comment:
setup won't wait for initialize to be complete if we do this. Could you use a completable future?

Reply:
Yep